dandi/bids_validator_deno/__init__.py

```diff
@@ -0,0 +1,445 @@
+        if config is not None:
+            # Write the config to a file in the temporary directory
+            configfile_path = Path(tmp_dir) / config_fname
+            configfile_path.write_text(json.dumps(config))
+
+            # Add the `--config` option
+            conditional_ops.extend(["--config", str(configfile_path)])
+
+        outfile_path = Path(tmp_dir) / out_fname
+        result = _invoke_validator(
+            ["--json", "--outfile", str(outfile_path), *conditional_ops, str(dir_)]
+        )
+        # Read the validation result from the outfile if it exists
+        outfile_content = outfile_path.read_text() if outfile_path.exists() else None
+
+    # The condition of this statement may need to change in the future.
+    # See https://github.com/bids-standard/bids-validator/issues/191 for details
+    if result.returncode not in range(0, 2) or result.stderr != "":
+        raise ValidatorError(
+            result.args,
+            result.returncode,
+            result.stdout,
+            result.stderr,
+            outfile_content,
+        )
+
+    assert (
+        outfile_content is not None
+    ), "`outfile_content` should not be None when validation is successful"
+
+    # Parse the content, in JSON format, of the outfile
+    return BidsValidationResult.model_validate_json(outfile_content, strict=True)
+
+
+def get_version() -> str:
+    """
+    Return the version of the deno-compiled BIDS validator
+
+    Returns
+    -------
+    str
+        The version of the deno-compiled BIDS validator
+    """
+    return version(DISTRIBUTION_NAME)
+
+
+def _harmonize(
+    bv_result: BidsValidationResult, ds_path: Path
+) -> list[ValidationResult]:
+    """
+    Harmonize a `BidsValidationResult` object into a list of `ValidationResult` objects
+
+    Parameters
+    ----------
+    bv_result : BidsValidationResult
+        The `BidsValidationResult` object representing the result of the validation
+        using the deno-compiled BIDS validator
+    ds_path : Path
+        The path to the dataset that has been validated to produce the `bv_result`
+        object
+
+    Returns
+    -------
+    list[ValidationResult]
+        A list of `ValidationResult` objects in which each object represents an issue
+        in the validation result.
+    """
+    # Ensure the path is absolute and in canonical form
+    ds_path = ds_path.resolve()
+
+    issues = bv_result.issues.issues
+    code_messages = bv_result.issues.codeMessages
+    schema_version = bv_result.summary.schemaVersion
+    dandiset_path = find_parent_directory_containing("dandiset.yaml", ds_path)
+
+    origin = Origin(
+        type=OriginType.VALIDATION,
+        validator=Validator.bids_validator_deno,
+        validator_version=get_version(),
+        standard=Standard.BIDS,
+        # todo: the BIDSVersion is unavailable through the validator;
+        #   However, the schema version is actually more precise
+        standard_version=schema_version,
```

Comment:
unfortunately it is not the standard version, so we should not report schema here... There is none "known" from validator ATM, but we can get it from dataset_description.json, although it would not be the version against which it was validated, so it is all not yet "clear". Let's mention related issue and make it more "explicit" for now:

```suggestion
        # TODO: the BIDSVersion is unavailable through the validator,
        #   see https://github.com/bids-standard/bids-validator/issues/10
        #   However, we know the schema version, which relates 1-to-1, so let's report that for now:
        standard_version=f"schema:{schema_version}",
```
but this raises the question also "version of BIDS in dataset" vs "version validated against"... dunno what to do about that for now.

Reply:
or may be let's be "explicit" and reserve also `standard_schema_version` and store this version there?

Reply:
> but this raises the question also "version of BIDS in dataset" vs "version validated against"... dunno what to do about that for now.

I think `standard_version` in `ValidationResult.origin` should always be set to `"version validated against"` and possibly that's all that is needed.

Reply:
`standard_schema_version` is now used to store the schema version, and `standard_version` is left to be `None` for now.